Source/WebKit/GPUProcess/graphics/WebGPU/RemoteCommandEncoder.cpp

```diff
@@ -48,7 +48,15 @@ RemoteCommandEncoder::RemoteCommandEncoder(PAL::WebGPU::CommandEncoder& commandE
     m_streamConnection->startReceivingMessages(*this, Messages::RemoteCommandEncoder::messageReceiverName(), m_identifier.toUInt64());
 }
 
-RemoteCommandEncoder::~RemoteCommandEncoder() = default;
+RemoteCommandEncoder::~RemoteCommandEncoder()
+{
+    destroy();
+}
+
+void RemoteCommandEncoder::destroy()
+{
+    m_objectHeap.removeObject(m_identifier);
+}
```

Comment:
Still confused about this. The destroy() message will be received, then we'll remove the object from the object heap, then that will cause the object's destructor to be called, which will cause destroy() again?